File: brightlearn-backend/src/main/java/com/brightlearn/service/ReviewService.java
package com.brightlearn.service;

import com.brightlearn.entity.Course;
import com.brightlearn.repository.CourseRepository;
import com.brightlearn.repository.EnrollmentRepository;
import com.brightlearn.entity.Review;
import com.brightlearn.repository.ReviewRepository;
import com.brightlearn.entity.User;
import com.brightlearn.repository.UserRepository;

import org.springframework.stereotype.Service;

@Service
public class ReviewService {

    private final ReviewRepository reviewRepository;
    private final EnrollmentRepository enrollmentRepository;
    private final UserRepository userRepository;
    private final CourseRepository courseRepository;

    public ReviewService(
            ReviewRepository reviewRepository,
            EnrollmentRepository enrollmentRepository,
            UserRepository userRepository,
            CourseRepository courseRepository) {
        this.reviewRepository = reviewRepository;
        this.enrollmentRepository = enrollmentRepository;
        this.userRepository = userRepository;
        this.courseRepository = courseRepository;
    }

    public void submitReview(Long courseId, String username, int rating, String comment) {

        User student = userRepository.findByUsername(username).orElseThrow();
        Course course = courseRepository.findById(courseId).orElseThrow();

        if (!enrollmentRepository.existsByStudentIdAndCourseId(student.getId(), courseId)) {
            throw new RuntimeException("You must enroll before reviewing");
        }

        reviewRepository.save(new Review(student, course, rating, comment));
    }
}

File: brightlearn-frontend/src/services/reviewService.ts
import api from './api';
import { Review } from '../types';

export const reviewService = {
  // Submit review: rating (int 1-5) and comment as query params
  submitReview: (courseId: number, rating: number, comment: string) =>
    api.post(`/reviews/${courseId}`, null, { params: { rating, comment } }),

  getCourseReviews: (courseId: number) =>
    api.get<Review[]>(`/reviews/course/${courseId}`),
};
